clients/egui/src/account/mod.rs

```diff
@@ -703,6 +711,44 @@ impl AccountScreen {
             update_tx.send(AccountUpdate::FileCreated(result)).unwrap();
         });
     }
+    fn create_file(&mut self) {
+        let mut focused_parent = self.tree.root.file.id;
+        for id in self.tree.state.selected.drain() {
+            focused_parent = id;
+        }
+        let core = self.core.clone();
+        let update_tx = self.update_tx.clone();
+        thread::spawn(move || {
+            let mut max = NameComponents::from("untitled-0.md");
+
+            let focused_parent = core.get_file_by_id(focused_parent).unwrap();
+            let focused_parent = if focused_parent.file_type == FileType::Document {
+                focused_parent.parent
+            } else {
+                focused_parent.id
+            };
+            core.get_children(focused_parent)
+                .unwrap()
+                .iter()
+                .for_each(|f| {
+                    let nc = NameComponents::from(&f.name);
+                    if let Some(variant) = nc.variant {
+                        if variant > max.variant.unwrap_or_default() {
+                            max = nc;
+                        }
+                    }
+                });
+
+            let result = core
+                .create_file(
+                    max.generate_next().to_name().as_str(),
+                    focused_parent,
+                    lb::FileType::Document,
+                )
+                .map_err(|err| format!("{:?}", err));
+            update_tx.send(AccountUpdate::FileCreated(result)).unwrap();
```

Comment:
I don't think this is the right strategy. Files with different prefixes will increment the variant. So if you have `image-conflict-3.png` as the only file in a folder you're going to create a file called `untitled-3.md`. You're also never going to create a file called `untitled-0.md`.

IMO the strategy should be something like:
start with `untitled.md`, check for conflicts with siblings, until there is no conflict continue to generate the `.next()`.

Reply:
okay i fixed it so that only file names `untitled` are considered. 

here's the proposed procedure: 
1. filter children of parent and collect a vector of files whose name is `untitled`
2. sort the vector based on variant 
3. traverse the vector until you find a gap for insertion either in the start, middle, or end. for example. 
   - `1,2,3,4` the gap is at the start, create `untitled.md`
   - `0,1,2,4` the gap is after the 2, create `untitled-3.md`